### DISCUSSION TITLE
Is it worth porting a native Android App to Maui?

### DISCUSSION THREAD

**Discussion by ronenfe** [dotnet/maui#21504] [Q&A]
I am considering porting my native android app to Maui

- I need to save settings on exit and load settings on start.
- Foreground service which receives location updates and broadcast them to the main activity.
- A circular slider control like this: https://github.com/bikcrum/CircularRangeSlider-Android, this is not critical, if I can use maybe a range slider instead.
- Control the volume of the device from the app.

Does it worth porting it to Maui so later I can make also an IOS app or will it be too difficult and it will be easier to develop it on Android and iOS separately?

**Comment by JohnHDev** [comment]
It depends on how confident you are working with C#, .NET MAUI (and Xamarin Forms before it) can offer a massive productivity boost if you are, but you also need to be aware it isn't perfect. You could put together a prototype to see if it suits your use cases.

**Comment by tschbc** [comment]
I recommend making a proof of concept app that satisfies your requirements at a base level so you can explore what MAUI can do and how it does it.

I've found that when things work correctly in MAUI, they tend to work _very well_. When they don't work (which unfortunately happens a lot, even with basic things like Editors and Layouts) you'll often be scrambling for a workaround which may not always be feasible. For the past year I've worked with MAUI, I've probably spent around 10-15% of my time just working around MAUI issues. 

I know other frameworks have their issues too, but I don't remember ever having to workaround a framework issue when I used Ionic & Cordova in the past, for example. I've certainly never needed to browse a framework's issues page on GitHub before I used MAUI.

**Reply by ronenfe** [reply]
It can happen easily and more with android studio bugs. The thing is I also didn't find documentation and code samples for how to use those features same as I could find for native development.

**Reply by tschbc** [reply]
For device features you'll mainly be interested in the [Platform integration](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/?view=net-maui-8.0) documentation ([Preferences](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/storage/preferences?view=net-maui-8.0&tabs=android) is the MAUI API to interact with SharedPreferences and their equivalents on other platforms)

Despite my issues with MAUI itself I've found that if documentation exists for something I'm usually satisfied with the amount of detail available.

I also haven't explored it much, but there is a https://github.com/davidortinau/maui-samples repo available.
